### PR TITLE
For #268: Implementing Plugin upgrade() and push() methods

### PR DESCRIPTION
**Pull request by bkuzmic**
For #268:

- Implementing `upgrade()` and `push()` methods
- Added new class `ArrayPayloadOf` to extract Json request as `Iterator<JsonObject>`
- Added new puzzles: to make tests for `ArrayPayloadOf` and to finish Plugin's `configure()` method

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#280](https://github.com/amihaiemil/docker-java-api/issues/280) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#280](https://github.com/amihaiemil/docker-java-api/issues/280) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/4364fd21-b057-4619-861c-61052c026b7c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/amihaiemil/docker/ArrayPayloadOf.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.apache.http.HttpRequest;
+
+/**
+ * Json Array payload of an HttpRequest.
+ *
+ * @author Boris Kuzmic (boris.kuzmic@gmail.com)
+ * @since 0.0.8
+ * @todo #266:30min Add tests for ArrayPayloadOf.
+ */
+final class ArrayPayloadOf implements Iterator<JsonObject> {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/ArrayPayloadOf.java]: @bkuzmic It seems we are using this class only for test purposes? Can you put it in ``src/test/java`` together with others, like ``Condition`` and so? :D

Comment (bkuzmic) [reply src/main/java/com/amihaiemil/docker/ArrayPayloadOf.java]: @amihaiemil Sure, no problem.

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job was finished in 19 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31):

**Comment by 0crat** [comment]
The job [#280](https://github.com/amihaiemil/docker-java-api/issues/280) is now out of scope

**Comment by 0crat** [comment]
Job `gh:amihaiemil/docker-java-api#280` doesn't exist in WBS, can't create order

**Comment by 0crat** [comment]
Job `gh:amihaiemil/docker-java-api#280` doesn't exist in WBS, can't create order

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**